### ISSUE TITLE
Consider introducing canThrow experimental feature

### ISSUE THREAD

**Issue by bjornregnell** [lunduniversity/introprog#588]
Should we teach a bit about the new experimental canThrow?
See: 

https://github.com/lampepfl/dotty/blob/b65958f139a0a818d36ee04468ada93b8fb472ac/docs/docs/reference/experimental/canthrow.md

A good explanation of several problems with Java's checked exceptions:
https://www.artima.com/articles/the-trouble-with-checked-exceptions

**Comment by Husenap** [comment]
I think this would be a really nice feature to discuss in Chapter 11 along with contextual abstractions 🚀

**Comment by bjornregnell** [comment]
Yeah, I guess it's too early in w06 when we discuss exceptions. Perhaps the problem with (un)checked exceptions can be discussed in w06 and then the new (experimental) solution shown in w11. Contextual abstractions are cool and given--using is nice, but context functions ?=> are really challenging...  So its a delicate matter of deciding how much and how deep we should take this into basic exercises.

**Comment by fritjof-b** [comment]
I agree. However, I believe we could mention `saferExceptions` when discussing regular exceptions. Either as an 'advanced topic' or as the general outline of the above article. I don't see a reason in going too deep into the gritty details of `saferExceptions` early. The general use should be easy enough to understand once you grasp exceptions. As long as we supply sufficient and clear examples in how they are used.

**Comment by bjornregnell** [comment]
Seems reasonable. I'll check that out when I get to the lecture preparations before w06.
